resources.md

```diff
@@ -32,8 +32,26 @@ It's often said that the best way to learn to write programs is to write program
+
+- [Automate the Boring Stuff With Python](https://automatetheboringstuff.com/) by Al Sweigart. A practical guide to the sorts of things that you can do with Python, how to do them, and why to do them that way. Recommended for those who have some understanding of Python and want to apply and improve their knowledge. Experienced programmers without previous Python exposure will probably be able to make good use of this book as well. The adventurous beginner might be able to make reasonable progress using the introductory chapter to learn Python fundamentals, but this is probably not an ideal place to start your Python journey. The book's content is available for free on the website. Libraries: [MLN](https://find.minlib.net/iii/encore/record/C__Rb3867472), [MLN <small>1<sup>st</sup> ed</small>](https://find.minlib.net/iii/encore/record/C__Rb3226226), [BPL <small>1<sup>st</sup> ed</small>](https://bpl.bibliocommons.com/item/show/6969316075).
+
+- [Cracking Codes With Python](https://inventwithpython.com/cracking/) by Al Sweigart. Probably less practical than the author's book on automation, this book is a concrete review of Python programming techniques through an exploration of basic cryptography. Recommended for the advanced beginner interested in dipping their toe into the world of codes and information. This book will *not* bring you up to speed with modern crypto, which is extremely math-intensive, but it shows how to write programs to solve well-defined problems. The book's content is available for free on the website. Libraries: [MLN](https://find.minlib.net/iii/encore/record/C__Rb3685466).
+
+- [Impractical Python Projects](https://nostarch.com/impracticalpythonprojects) by Lee Vaughan. Somewhat more advanced than Sweigart's books, this book goes deep into Python libraries to do interesting and completely impractical things, such as using NLTK to write haikus, using PyGame to simulate alien volcanoes, or using Pillow to improve your astrophotography. Recommended for the more advanced beginner. If you're comfortable writing standard Python, this book will show you how to use other people's libraries to do cool things easily. Libraries: [MLN](https://find.minlib.net/iii/encore/record/C__Rb3725372), [BPL](https://bpl.bibliocommons.com/item/show/6971575075).
+
+- [Think Stats!](https://greenteapress.com/wp/think-stats-2e/) and [Think Bayes!](http://greenteapress.com/wp/think-bayes/) by Allen Downey. These books are aimed at teaching probability & statistics and Bayesian statistics, using Python rather than traditional mathematical expressions to make the concepts concrete. Recommended for more advanced beginners with an interest in the mathematical underpinnings of statistics. The website for both books has free PDF and HTML versions.
```

Comment:
I didn't include the library link for Think Stats, because it's for the first edition, which "did not use pandas, SciPy, or StatsModels". Given that it's free online, I'm inclined to steer folks to that.